### PR TITLE
CIAB: Make the socks container optional

### PR DESCRIPTION
**Pull request by JBevillC**
#### What does this PR do?

CDN-IN-A-Box Only
1) Moves the container "client" into the optional directory
2) Renames the container from "client" to  "socksproxy" 
3) Add "socksproxy.infra.ciab.test" A/AAAA record(s)  to ipv4/ipv6 zone files in the dns bind container.

Fixes #(issue_number)

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [X] Other _CDN-IN-A-Box________

#### What is the best way to verify this PR?

1) Start up the CIAB with the optional socksproxy container

```
# From $SRC_ROOT/infrastructure/cdn-in-a-box
alias odc='docker-compose -f docker-compose.yml -f optional/docker-compose.socksproxy.yml'
odc build 
odc kill && vncdc rm -fv 
docker volume prune -f
odc up
```

2) Configure browser to use the socks proxy
- Install the Foxy Proxy browser plugin.
- Add New Proxy, Manual Proxy Configuration, Host: localhost, Port: 9080, Check 'SOCKS Proxy', Select 'SOCKS v5'.
- Enable 'pre-defined and patterns' mode.

3) Type in the URL with your browser: 
- URL: https://trafficportal.infra.ciab.test

4) Note: For SSL validation to work, the CIAB SSL CA Certificate in traffic_ops/ca will need to be trusted by browser/system. 

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2895/
Test FAILed.

**Comment by JBevillC** [comment]
Updated Docs and rebased to master

File: docs/source/admin/quick_howto/ciab.rst [hunk 1]
```diff
@@ -109,16 +171,96 @@ The "enroller" provides an efficient way for Traffic Ops to be populated with da
...
+* Docker Host: http://localhost:9200/index.html
+
+.. _ciab-optional-containers:
+
+Optional Containers
+===================
+
+All optional containers that are not part of the core CDN-in-a-Box stack are located in the ``infrastructure/cdn-in-a-box/optional`` directory.
+
+.. code-block:: shell
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: Maybe this should be `text` syntax? It's not really valid shell commands, and I don't think any highlighting really makes sense.

File: docs/source/admin/quick_howto/ciab.rst [hunk 2]
```diff
@@ -99,6 +99,68 @@ variables.env
...
+   | CIAB-CA.crt               | Shared CA Certificate             | N/A                          |
+   +---------------------------+-----------------------------------+------------------------------+
+   | infra.ciab.test.crt       | Infrastruture Certificate         | \*.infra.ciab.test           |
+   +---------------------------+-----------------------------------+------------------------------+
+   | demo1.mycdn.ciab.test.crt | Demo1 Delivery Service Certificate| \*.demo1.mycdn.ciab.test     |
+   +---------------------------+-----------------------------------+------------------------------+
+   | demo2.mycdn.ciab.test.crt | Demo2 Delivery Service Certificate| \*.demo2.mycdn.ciab.test     |
+   +---------------------------+-----------------------------------+------------------------------+
+   | demo3.mycdn.ciab.test.crt | Demo3 Delivery Service Certificate| \*.demo3.mycdn.ciab.test     |
+   +---------------------------+-----------------------------------+------------------------------+
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: This indentation is inconsistent - above this point tabs were used. Can you either change the existing indentation to spaces or change this to tabs? I guess it doesn't matter which, as long as it's consistent, but for the record I think tabs are the superior choice.

File: docs/source/admin/quick_howto/index.rst [hunk 3]
```diff
@@ -27,4 +27,4 @@ Traffic Control is a complicated system, and documenting it is not trivial. Some
 	dnssec
 	ds_requests
 	steering
-	ciab
+        ciab
```
Comment (dangogh) [inline docs/source/admin/quick_howto/index.rst]: is that indent change intentional?

File: docs/source/admin/quick_howto/ciab.rst (hunk 2 above)
Comment (dangogh) [reply docs/source/admin/quick_howto/ciab.rst]: +1 on keeping existing indentation and making this consistent with it

File: docs/source/admin/quick_howto/ciab.rst [hunk 4]
```diff
@@ -99,19 +99,162 @@ variables.env
...
+
+.. [4] The ``ca`` volume is not purged with normal ``docker volume`` commands. This feature is by design to allow the existing shared SSL certificate to be trusted at the system level across restarts. To re-generate all SSL certificates and keys, remove the ``infrastructure/cdn-in-a-box/traffic_ops/ca`` directory before startup.
+
+Trusting the CA 
+---------------
+For developer and testing use-cases, it may be necessary to have full x509 CA validation by HTTPS clients [5]_.  For x509 validation to work properly, the self-signed x509 CA certificate must be trusted either at the system leevel or by the client applicatoin itself. Procedures to import and trust the CA x.509 certifcate are outlined below for OSX, Windows, and two Linuxs below [6]_.
```
[the remark this answers is not in the document]
Comment (dangogh) [reply docs/source/admin/quick_howto/ciab.rst]: +1 on not naming OS's here at all..

File: docs/source/admin/quick_howto/index.rst (hunk 3 above)
Comment (JBevillC) [reply docs/source/admin/quick_howto/index.rst]: Not intentional.  My .vimrc has a lot of custom stuff in it and was trying to reformat only part of the file. I'll fix and repush this

File: docs/source/admin/quick_howto/ciab.rst (hunk 2 above)
Comment (JBevillC) [reply docs/source/admin/quick_howto/ciab.rst]: Sounds good.  I'll update my .vimrc to not replace tabs with spaces.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2896/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2897/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2902/
Test PASSed.

File: docs/source/admin/quick_howto/ciab.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (JBevillC) [reply docs/source/admin/quick_howto/ciab.rst]: I've removed the "for OSX, Windows, and two Linuxs below" part of the paragraph.

[the remark this answers is not in the document]
Comment (JBevillC) [reply docs/source/admin/quick_howto/ciab.rst]: I really hate writing docs. can you tell?

File: docs/source/admin/quick_howto/ciab.rst [hunk 5]
```diff
@@ -99,19 +99,162 @@ variables.env
...
+
+.. [4] The ``ca`` volume is not purged with normal ``docker volume`` commands. This feature is by design to allow the existing shared SSL certificate to be trusted at the system level across restarts. To re-generate all SSL certificates and keys, remove the ``infrastructure/cdn-in-a-box/traffic_ops/ca`` directory before startup.
+
+Trusting the CA 
+---------------
+For developer and testing use-cases, it may be necessary to have full x509 CA validation by HTTPS clients [5]_.  For x509 validation to work properly, the self-signed x509 CA certificate must be trusted either at the system leevel or by the client applicatoin itself. Procedures to import and trust the CA x.509 certifcate are outlined below for OSX, Windows, and two Linuxs below [6]_.  
+
+Importing the CA Certificate on OSX
+-----------------------------------
+1. Copy the CIAB root CA certificate from ``infrastructure/cdn-in-a-box/traffic_ops/ca/CIAB-CA.crt`` to the Mac.
```
[the remark this answers is not in the document]
Comment (JBevillC) [reply docs/source/admin/quick_howto/ciab.rst]: Feel free to update it once it is merged.

File: docs/source/admin/quick_howto/ciab.rst (hunk 1 above)
Comment (JBevillC) [reply docs/source/admin/quick_howto/ciab.rst]: Feel free to update it once it is merged.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2905/
Test FAILed.

**Comment by dangogh** [comment]
this was included in #3005 -- closing this one to avoid merge conflicts

**Pull request closed without merging**